### ISSUE TITLE
Obscure error messaging when running script with --verify

### ISSUE THREAD

**Issue by 0xPhaze** [foundry-rs/foundry#3242]
### Component

Forge

### Have you ensured that all of these are up to date?

- [X] Foundry
- [X] Foundryup

### What version of Foundry are you on?

forge 0.2.0 (359dd77 2022-09-15T00:04:51.833948Z)

### What command(s) is the bug in?

forge script

### Operating System

macOS (Apple Silicon)

### Describe the bug

I was getting this weird error message.
![image](https://user-images.githubusercontent.com/103113487/190814702-64b5c7c4-cdf9-4383-b2cc-195d0a9e2bdc.png)

It took me a while to pin it down to using `forge script --verify` while broadcasting, but not generating any transactions. Removing the `--verify` allowed me to see the message: "no transactions generated"

**Comment by mattsse** [comment]
looks like the linking step is the root cause for this, are you using only libraries that are in the project or something at a specific address?

**Comment by 0xPhaze** [comment]
I'm not using any libraries with public functions. I'm working with already deployed contracts in my script, forking from mumbai. The `--verify` might have been misleading, because now, after running it again it gets stuck with or without.

**Comment by mattsse** [comment]
do you have the `libraries = []` key set in your config

https://book.getfoundry.sh/reference/config/solidity-compiler#libraries

**Comment by 0xPhaze** [comment]
That did the trick for me, thank you very much. It was previously just not present in the config.

**Comment by mattsse** [comment]
I'll keep this open to improve the error message(s)

**Comment by 0xPhaze** [comment]
Interestingly it appeared again. I played around a bit again and removed the `--verify` and it then worked (but without verifying the contracts of course).

**Comment by mattsse** [comment]
hmm, will try to reproduce this,

is this public by any chance?

**Comment by 0xPhaze** [comment]
Yes, here: https://github.com/0xPhaze/gang-war
It seems to be reproducible on anvil
```
forge script deploy --rpc-url "http://127.0.0.1:8545" --private-key 0x83631fd2da5250fa4f9dd8c1057f6e53126de6dd9274b9aefbcc3eae3ada6a5f -vvv --ffi --broadcast
```
Though because of the setup scripts, it will think it has successfully deployed and you'll have to run `rm -rf deployments/31337` to delete the stored deployments and "reset" it.

**Comment by 0xPhaze** [comment]
Following up here. This seems to occur when I add more functions (or any code). If I delete some code, I can deploy without an error (and I'm pretty sure it does not have anything to do with "linking", unless I misunderstand that word). Could this then simply be when the max code size is reached? Did the error messages get messed up?

**Comment by 0xPhaze** [comment]
Fixed in gakonst/ethers-rs#1743
